### PR TITLE
op-node: Add unit tests to origin selector

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Adds some unit tests.

### CODE REVIEW

**Comment by trianglesphere** [comment]
FYI it's op-e2e which is flaking rather than the added tests

**Comment by tynes** [comment]
The devnet failed by timeout even though it was making progress, it seemed like the output proposer was really slow? It was waiting for an output proposal to be posted

File: op-node/rollup/driver/origin_selector_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/testlog"
+	"github.com/ethereum-optimism/optimism/op-node/testutils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestOriginSelectorAvances ensures that the origin selector
```
Comment (tynes) [inline op-node/rollup/driver/origin_selector_test.go:16]: typo in advances

**Comment by trianglesphere** [comment]
> The devnet failed by timeout even though it was making progress, it seemed like the output proposer was really slow? It was waiting for an output proposal to be posted

Weird. In the devnet we should be using the faster version of the proposer (that does not wait on L1 finalization)

https://github.com/ethereum-optimism/optimism/commit/05d367e5680ac010ad0a016f41047b1eb768032d#diff-4911e38c0c2bfa62bebb0161d1ff90483d490d82e945a676cc56205e34f50345

**Comment by tynes** [comment]
It would be a lot easier to verify the validity of these tests if there were comments describing how the values used enable the desired test scenario

**Comment by trianglesphere** [comment]
> It would be a lot easier to verify the validity of these tests if there were comments describing how the values used enable the desired test scenario

added comments about this

**Review by tynes — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Pull request merged**